### PR TITLE
PR: Add plugin actions to the 'NotebookWidget'

### LINKED ISSUE #31: Add plugin actions (Open, Save, etc) to the `NotebookWidget` context menu
The idea is that by doing a right-click on any notebook, users can get access to those actions instead of going to the Options plugin menu.

### PR DESCRIPTION
**Pull request by dalthviz**
Fixes #31 

A preview:

actions

### CODE REVIEW

File: spyder_notebook/widgets/dom.py [hunk 1]
```diff
@@ -18,10 +18,7 @@ class DOMWidget(WebView):
 
     def __init__(self, parent):
         super(DOMWidget, self).__init__(parent)
-        if WEBENGINE:
-            self.page = self.page()
-        else:
-            self.page = self.page().mainFrame()
```
Comment (ccordoba12) [inline spyder_notebook/widgets/dom.py]: Please don't remove this. This is very important to make things work for the packages provided by Anaconda and the PyQt5 wheels.

File: spyder_notebook/widgets/dom.py [hunk 2]
```diff
@@ -30,7 +27,7 @@ def evaluate(self, script):
         if WEBENGINE:
             return self.page.runJavaScript("%s" % script)
         else:
-            return self.page.evaluateJavaScript("%s" % script)
```
Comment (ccordoba12) [inline spyder_notebook/widgets/dom.py]: Revert these changes too.

File: spyder_notebook/widgets/dom.py (hunk 1 above)
Comment (ccordoba12) [inline spyder_notebook/widgets/dom.py]: Ok, now I see what's the problem: I made the `page` method an attribute of this widget, and that break everything!! My bad, sorry ;-)

Then let's change this to

```python
if WEBENGINE:
    self.dom = self.page()
else:
    self.dom = self.page().mainFrame()
```

File: spyder_notebook/widgets/dom.py (hunk 2 above)
Comment (ccordoba12) [inline spyder_notebook/widgets/dom.py]: Please also change the use of `self.page` here to `self.dom`

File: spyder_notebook/widgets/client.py [hunk 3]
```diff
@@ -58,11 +58,13 @@ def contextMenuEvent(self, event):
...
-        actions = [self.pageAction(QWebEnginePage.SelectAll),
-                   self.pageAction(QWebEnginePage.Copy), None,
-                   self.zoom_in_action, self.zoom_out_action]
+        plugin_actions = self.parent().plugin_actions
+        actions = plugin_actions + [None,
+                                    self.pageAction(QWebEnginePage.SelectAll),
+                                    self.pageAction(QWebEnginePage.Copy), None,
+                                    self.zoom_in_action, self.zoom_out_action]
         if not WEBENGINE:
-            settings = self.page().settings()
```
Comment (ccordoba12) [inline spyder_notebook/widgets/client.py]: After making the changes I suggested to `DOMWidget`, you can restore this line to its previous state.

File: spyder_notebook/widgets/dom.py (hunk 2 above)
Comment (ccordoba12) [inline spyder_notebook/widgets/dom.py]: And leave things as they were :-)

Comment (goanpeca) [inline spyder_notebook/widgets/dom.py]: Could we use str.format() instead of %, it feels like C

Comment (ccordoba12) [inline spyder_notebook/widgets/dom.py]: Oh, I took that from the Ghost.py project and leave it like as it was. But there's no problem in using `format`.

@dalthviz, please also make @goanpeca's suggested change :-)

**Comment by ccordoba12** [comment]
This looks good to me now. Thanks @dalthviz!
